Backend/app/provider-platform/static-offer-driver-app/Main/src/Domain/Action/Dashboard/Ride.hs

```diff
@@ -308,6 +308,73 @@ syncCompletedRide ride booking = do
     CallBAP.sendRideCompletedUpdateToBAP booking ride fareBreakup
   pure $ Common.RideSyncRes Common.RIDE_COMPLETED "Success. Sent ride completed update to bap"
 
+---------------------------------------------------------------------
+
+multipleRideSync :: ShortId DM.Merchant -> Common.MultipleRideSyncReq -> Flow Common.MultipleRideSyncRes
+multipleRideSync merchantShortId rideSyncReq = do
+  merchant <- findMerchantByShortId merchantShortId
+  let rideIds = map (cast @Common.Ride @DRide.Ride) rideSyncReq.rideIds
+
+  rides <- QRide.findByAllRideIds rideIds
+  bookings <- QBooking.findAllByBookingIds $ map (.bookingId) rides
```

Comment:
But what if list order of `rides`, `bookings` and `rideIds` will be not the same? It definitely will be not the same. Then `ridesBookingsZip` will contain not corresponding data. I think it would be better to fetch all data and also check merchant access within one join query:
`ridesBookingsZip <- findAllRidesBookingsByRideId rideIds merchantId`

Reply:
fixed